### ISSUE TITLE
Fyne API storage implement io.Seeker (mainly for mobile)

### ISSUE THREAD

**Issue by alexballas** [fyne-io/fyne#2352]
### Is your feature request related to a problem? Please describe:
Hello, In my attempt to use https://pkg.go.dev/net/http#ServeContent I noticed that the fyne storage api methods only support ReadCloser https://developer.fyne.io/api/v2.0/storage/. ServeContent expects an io.ReadSeeker which *os.File implementes already.
This came as a need when I tried to read files in an Android device and serve them with http.ServeContent since the regular os file methods (https://pkg.go.dev/os) wont work there.

**Comment by andydotxyz** [comment]
Nice idea. We can't do it for all as many data sources are not seekable, but it should be possible to extend for file based operations that support it.

**Comment by Jacalz** [comment]
Isn't this technically a duplicate of https://github.com/fyne-io/fyne/issues/1164? It was closed a long time ago as being not possible. However, this is still very high on my list of requested features so I would not be against leaving this open instead if we decide that this is possible to implement.

**Comment by andydotxyz** [comment]
These seem different to me. The previous request was that all items are seekable, which is not possible. Adding seek to the file repos implementation with some optional abstraction on top should be possible.

**Comment by Jacalz** [comment]
The fundamental idea seems to be exactly the same to me. I think it's just a matter of different wording.

**Comment by andydotxyz** [comment]
I'd say the last sentence in the old issue wraps it up "I do know that not all URIs will be files, but I cannot see much reason for not selecting a file in a file picker ;)".

The file picker is not limited to local files that can be seeked, and so we cannot require that all items returned by a file picker are seekable. But we can make it an optional extra that some apps could require additionally to selecting the more generic concept of a data source represented by file icons.

**Comment by andydotxyz** [comment]
For a more concrete illustration consider HTTP - accessing a file in that manner will only be seekable if the server supports the `Range` headers. We can't require that all expose seek support when the source data may not allow such operations.

**Comment by Jacalz** [comment]
Yeah. That issue is very old and created way before a lot of knowledge that I have picked up before. All I am trying to say was that I wanted io.ReadSeeker in some way but I suggested for all because I didn't know better at the time :)

**Comment by andydotxyz** [comment]
That's fair enough. I think you can understand why others not see it as exactly the same as the old feature request though?
Sounds like if we can solve this you'll both be happy though :).

**Comment by Jacalz** [comment]
> That's fair enough. I think you can understand why others not see it as exactly the same as the old feature request though?

Yeah, absolutely.

> Sounds like if we can solve this you'll both be happy though :).

:)

**Comment by bon-ami** [comment]
Has this been abandoned?

**Comment by Jacalz** [comment]
> Has this been abandoned?

No. We don’t really abandon things in that way. If the issue is open still then no one has had time to work on it yet

**Comment by alexballas** [comment]
I know this is an ancient issue. For anyone facing this, I just ended up copying the data to a temp file and use that for the seek or any other file operations. Afterwards I just remove it.
